### PR TITLE
Half working left or right toolbar button pos

### PR DESCRIPTION
**Pull request by Jacalz**
This is unfortunately as far as I could get on my own. There are various issues that I have been unable to solve.

- The windows do not resize correctly when using right position.
- Already open windows do not update when applying the setting in settings.

- The correct selected values are not shown in settings on start. Seems to be an issue in the clock as well so not related to this PR, I guess.

### CODE REVIEW

File: wm/border.go [hunk 1]
```diff
@@ -53,20 +53,42 @@ func NewBorder(win fynedesk.Window, icon fyne.Resource, canMaximize bool) *Borde
 	}}
 
 	title := canvas.NewText(win.Properties().Title(), theme.ForegroundColor())
-	titleBar := newColoredHBox(win.Focused(), win, makeFiller(0),
-		newCloseButton(win),
-		max,
-		min,
-		title,
-		layout.NewSpacer())
+	buttonPos := fyne.CurrentApp().Preferences().StringWithFallback("toolbarbuttonposition", "Left")
```
Comment (Jacalz) [inline wm/border.go:56]: I could not really figure out how to get this working using the settings code. Any advice would be appreciated.

**Comment by andydotxyz** [comment]
I guess this should be a draft PR for now?
I'll see if I can look into the settings code.

File: wm/border.go (hunk 1 above)
Comment (andydotxyz) [reply wm/border.go:56]: This looks OK - what part of it isn't working as you'd like?

**Comment by andydotxyz** [comment]
I can't see anything wrong, can you expand on the problem you were encountering @Jacalz ?

File: wm/border.go (hunk 1 above)
Comment (Jacalz) [reply wm/border.go:56]: I just thought that I'd be able to use the actually DeskSettings struct, but if this is the good way to go, I'm happy with that.

**Comment by Jacalz** [comment]
> I can't see anything wrong, can you expand on the problem you were encountering @Jacalz ?

Sure thing. The resizing of windows is entirely broken when using window decorations on the right side. I've tried to put the spacers in different places but to no avail. Having them on the left works as expected.  

https://user-images.githubusercontent.com/25466657/109161196-0f832a80-7777-11eb-9a1a-f600eec60769.mp4

There is of course also the issue with windows not updating when the left/right setting change. You have to open a new window for it to show.

**Comment by andydotxyz** [comment]
Ah yes I see. There is an optimisation in the code that chops the image in two so it can be resized without redrawing.
If you move the chop position in appropriately it should be fixed.

**Comment by andydotxyz** [comment]
See `internal/x11/win` around 345:

`iconAndBorderSizePix := iconSizePix + x11.BorderWidth(x11.XWin(f.client))*2`

**Comment by andydotxyz** [comment]
This should now be completed on `feature/decorations` branch for perusal before pulling in here/opening new PR.

**Comment by Jacalz** [comment]
Closing this in favor of a new PR coming up shortly...

**Pull request closed without merging**